index.php

```diff
@@ -63,6 +64,16 @@
 
 // Current locks.
 echo html_writer::tag('h1', get_string('h1_current', 'tool_lockstats') . ' (' . count($records) . ')');
+$sessionkey = sesskey();
+$url = new moodle_url("/admin/tool/lockstats/index.php", array('action' => 'releaseall'));
+$currentlocks = $current->get_current_locks();
+if ($action === 'releaseall') {
+    $result = releasealllocks($currentlocks, $sessionkey);
```

Comment:
It's generally better to split the 'web' code from the 'business logic' code. ie right here call 'require_sesskey' before you call releasealllocks, and then releasealllocks isn't tightly coupled to the web, ie it could be called from cron or cli or unit tests

Reply:
After releasing the locks, you must redirect to get rid of the sesskey.

I don't think the $result has that much value, I'd just ignore the result and redirect. As we are ignoring the value there is little point in returning it in the first place

Reply:
I am now using require_sesskey here before calling releasealllocks.